python/pylibcugraph/pylibcugraph/tests/test_node2vec.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import cupy as cp
+import numpy as np
+
+
+# =============================================================================
+# Test data
+# =============================================================================
+# The result names correspond to the datasets defined in conftest.py
+# Note: the only deterministic path(s) in the following datasets
+# are contained in Simple_1
+_test_data = {"karate.csv": {
+                  "seeds": cp.asarray([0, 0], dtype=np.int32),
+                  "paths": cp.asarray([0, 8, 33, 29, 26, 0, 1, 3, 13, 33],
```

Comment:
I think this is good for now but I am a little concern with this approach of testing because of the randomness of the results.
1) did you make sure there are only 2 paths with depth 5 starting from seed 0?
2) Even if it is the case, the testing seems hardcoded and only a few seeds are tested. 
An alternative could be to pick random seeds and verify the accuracy of the results by checking the predecessors.  To illustrate this, let's take the first path 
[0, 8, 33, 29, 26]. you verify that 0 is a predecessor of 8, then 8 being a predecessor of 33 and so on... but I am not sure how to easily do this without involving cugraph.

Reply:
This approach of testing was also in the cugraph level, but yes when this was done I did not check that the walks taken were in fact valid, just putting placeholder values and focusing on making sure the start of each path was each of the seeds. I've since made the testing more robust on the cugraph side, but for pylibcugraph it's more complicated, I'm still figuring that out right now but I can include a FIXME or add an issue to revisit later